pkg/controller/operators/catalog/operator.go

```diff
@@ -1121,6 +1136,14 @@ func (o *Operator) ensureSubscriptionCSVState(logger *logrus.Entry, sub *v1alpha
 	if err != nil {
 		logger.WithError(err).WithField("currentCSV", sub.Status.CurrentCSV).Debug("error fetching csv listed in subscription status")
 		out.Status.State = v1alpha1.SubscriptionStateUpgradePending
+		if failForwardEnabled {
+			ip, err := o.client.OperatorsV1alpha1().InstallPlans(sub.GetNamespace()).Get(context.TODO(), sub.Status.InstallPlanRef.Name, metav1.GetOptions{})
+			if err != nil {
+				logger.WithError(err).WithField("currentCSV", sub.Status.CurrentCSV).Debug("error fetching installplan listed in subscription status")
+			} else if ip.Status.Phase == v1alpha1.InstallPlanPhaseFailed {
+				out.Status.State = v1alpha1.SubscriptionStateFailed
+			}
```

Comment:
I'll preface this with I could be reading these changes wrong: let's say I have an initial catalog that contains a v1.0.0 that produces a successful install plan, then update that catalog with an olm.bundle v1.0.1 that replaces v1.0.0 that contains a bad image, and eventually patch that catalog to replace that bad version. What ends up happening here? 

It looks like we ensure the Subscription has an install plan reference between calling this ensureSubscriptionCSVState function, which has the following terminal state:

```go
if sub.Status.InstallPlanRef != nil || sub.Status.Install != nil {
		return sub, false, nil
}
```

I'd gather in this poor v1.0.0 -> v1.0.1 rollout scenarios, I'd have a Subscription that's pointing to the new, yet failed CSV that was unpacked, yet that CSV doesn't exist (e.g. the InstallPlan had a bad olm.bundle image). This would imply that despite the UpgradeFailed CSV state, we'd still have an InstallPlan reference that's present in the Subscription's status, which wouldn't result in a new resolution being produced. Is this the right read? Should we be hacking around the populated InstallPlan reference, and reseting that status field? Am I missing something obvious?

I can pull down these changes tomorrow and double check this potential edge case (although this feels like the happy for automatically rolling forward).